.github/workflows/ci.yml

```diff
@@ -51,47 +63,85 @@ jobs:
           V=${{ matrix.python-version }}
           DO_MYPY=1
 
-          if [[ "$V" = pypy-* ]]; then
+          if [[ "$V" == "3.7" || "$V" == "3.8" ]]; then
             DO_MYPY=0
-
-            V=pypy3
-          else
-            if [[ "$V" == "3.7" || "$V" == "3.8" ]]; then
-              DO_MYPY=0
-              echo "skipping $V"
-            else
-              echo "not skipping $V"
-            fi
-
-            V=py$(echo $V | tr -d .)
           fi
 
           echo DO_MYPY=$DO_MYPY >>$GITHUB_ENV
-          echo TOX_PYTHON=$V >>$GITHUB_ENV
+          echo TOX_PYTHON=py$(echo $V | tr -d .) >>$GITHUB_ENV
 
           python -Im pip install tox
 
       - run: python -Im pip install tox-uv
         if: matrix.python-version != '3.7'
 
-      - run: python -Im tox run -e ${{ env.TOX_PYTHON }}-tests
       - run: python -Im tox run -e ${{ env.TOX_PYTHON }}-mypy
         if: env.DO_MYPY == '1'
 
+      - name: Remove src to ensure tests run against wheel
+        run: rm -rf src
+
+      - run: >
+          python -Im
+          tox run
+          --installpkg dist/*.whl
+          -e ${{ env.TOX_PYTHON }}-tests
+
       - name: Upload coverage data
         uses: actions/upload-artifact@v4
         with:
           name: coverage-data-${{ matrix.python-version }}
           path: .coverage.*
           if-no-files-found: ignore
 
+  tests-pypy:
+    name: Tests on ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    needs: build-package
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - pypy-3.8
```

Comment:
Isn't 3.8 unsupported by PyPy already? Pytest was hitting terrible crashes related to a bug in their GC quite often. So we ended up upgrading to 3.9 that's the lowest supported today and has a fix for that obscure bug.

Reply:
If that’s true, I’m more than happy to drop it. I’ve just kept what was before

Reply:
“We currently support python 3.10, 3.9, and 2.7.” yay